android-core/src/main/java/com/mparticle/modernization/event_logging/MParticleEventLogging.kt

```diff
@@ -0,0 +1,14 @@
+package com.mparticle.modernization.event_logging
+
+import com.mparticle.BaseEvent
+import com.mparticle.modernization.MParticleComponent
+import java.math.BigDecimal
+
+interface MParticleEventLogging : MParticleComponent {
+    //All common, commerce, screen, pushRegistration, notification, notificationOpened and NetworkPerformance events should be logged with the same function
+    fun logEvent(event: BaseEvent)
+    fun leaveBreadcrumb(breadcrumb: String)
+    fun logError(message: String, params: Map<String, String>? = null)
+    fun logLtvIncrease(valueIncreased: BigDecimal, eventName: String? = null, params: Map<String, String>? = null)
```

Comment:
Lifetime value increase appears to be a special case of a commerce event. Any chance this could be collapsed into logEvent()?

Reply:
We could create the abstraction of LtvEvent extending from Base event. Here is where we need to choose, bigger inheritance tree and smaller api or the other way round. But is totally possible if we want to.